File: android/app/src/main/kotlin/model/WidgetDataModel.kt
package model

import com.google.gson.annotations.SerializedName


data class WidgetDataModel (
    @SerializedName("today") val today : List<Today>?,
    @SerializedName("item") val item : Item?,
    @SerializedName("fortune") val fortune : Fortune?
)

data class Fortune (
    @SerializedName("health") val health : Health?,
    @SerializedName("love") val love : Health?,
    @SerializedName("money") val money : Health?
)

data class Health (
    @SerializedName("name") val name : String? = "",
    @SerializedName("contents") val contents : String? = "0"
)

data class Item (
    @SerializedName("number") val number : Health?,
    @SerializedName("color") val color : Health?,
    @SerializedName("direction") val direction : Health?
)

data class Today (
    @SerializedName("word") val word : String? = "",
    @SerializedName("score") val score : String? = "0",
    @SerializedName("score_name") val scoreName : String? = ""
)

File: ios/HomeWidgetsV17/widgets/WidgetDataModel.swift
struct Today: Codable {
    let word: String?
    let score: String?
    let scoreName: String?
    
    enum CodingKeys: String, CodingKey {
        case word
        case score
        case scoreName = "score_name"
    }
}

struct Direction: Codable {
    let name: String?
    let contents: String?
}

struct Colors: Codable {
    let name: String
    let contents: String
}

struct Number: Codable {
    let name: String
    let contents: String
}

struct Item: Codable {
    let direction: Direction
    let color: Colors
    let number: Number
}

struct WidgetData: Codable {
    let today: [Today]?
    let item: Item?
    let fortune: Fortunes?
}

struct Love: Codable {
    let name: String
    let contents: String
}

struct Money: Codable {
    let name: String
    let contents: String
}

struct Health: Codable {
    let name: String
    let contents: String
}

struct Study: Codable {
    let name: String
    let contents: String
}

struct Work: Codable {
    let name: String
    let contents: String
}

struct Fortunes: Codable {
    let love: Love
    let money: Money
    let health: Health
    let study: Study
    let work: Work
}

